src/main/java/net/sf/jabref/model/ChainNode.java

```diff
@@ -126,6 +126,7 @@ protected void setParent(T parent) {
      *
      */
     public void removeChild() {
+        // NPE if this is ever called
         child.setParent(null);
```

Comment:
> Isn't it better that the author of the faulty code fixes it

The nice thing about open source code is that there is no "your / mine code", everyone is responsible for all the code. I can't count how many times I already fixed some small bugs which somebody else created and similarly many others (including you) cleaned up my mess.

Please just add the if statement there and then merge this in directly (ready-for-review removed for this reason).